roles/terrible/templates/terraform-vm.tf.j2

```diff
@@ -38,6 +38,8 @@ resource "libvirt_domain" "domain_{{ inventory_hostname }}" {
   memory = {{ memory }}
   vcpu   = {{ cpu }}
 
+  autostart = {% if vm_autostart %}true{% else %}false{% endif %}
```

Comment:
Magnific way to solve the problem.
We will not have any issue with other providers (like proxmox or ovirt) integration!

Reply:
This should not influence other providers, 
I think the right approach would be to have a `terraform-vm.tf.j2` for each provider supported, and simply called as `terraform-vm-{{ provider_name }}.tf.j2 ` for example.
I don't think that other providers would be too similar to this, so instead of creating one giant j2 file, it would be better to have a template for each provider. 
The most important thing is that we do not have to duplicate any code in ansible (using the provider name in the file-name for example) and that the concept of inventory as single source of truth is intact.